src/test/http_rest_api_handler_test.cpp

```diff
@@ -426,6 +470,175 @@ TEST_F(ConfigReload, StartWith1DummyThenReloadToAddPipeline) {
     EXPECT_EQ(status, ovms::StatusCode::OK_RELOADED);
 }
 
+static const char* configWithPipelineWithInvalidOutputs = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy"
+            }
+        }
+    ],
+    "pipeline_config_list": [
+        {
+            "name": "pipeline1Dummy",
+            "inputs": ["custom_dummy_input"],
+            "nodes": [
+                {
+                    "name": "dummyNode",
+                    "model_name": "dummy",
+                    "type": "DL model",
+                    "inputs": [
+                        {"b": {"node_name": "request",
+                            "data_item": "custom_dummy_input"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "a",
+                        "alias": "new_dummy_output"}
+                    ] 
+                }
+            ],
+            "outputs": [
+                {"custom_dummy_output": {"node_name": "dummyNode",
+                                        "data_item": "non_existing_output"}
+                }
+            ]
+        }
+    ]
+})";
+
+TEST_F(ConfigReload, StartWith1DummyThenReloadToAddPipelineWithInvalidOutputs) {
+    ModelManagerTest manager;
+    SetUpConfig(configWith1Dummy);
+
+    auto handler = ovms::HttpRestApiHandler(10);
+    std::string response;
+
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+    LoadConfig(manager);
+    RemoveConfig();
+    SetUpConfig(configWithPipelineWithInvalidOutputs);
+    std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
I think you can get rid of this sleep when using ConstructorEnabledModelManager.